src/clj/fluree/db/connection/system.cljc

```diff
@@ -0,0 +1,159 @@
+(ns fluree.db.connection.system
+  (:require [fluree.db.connection :as connection]
+            [fluree.db.cache :as cache]
+            [fluree.db.storage.file :as file-store]
+            [fluree.db.storage.memory :as memory-store]
+            [fluree.db.storage.remote :as remote-store]
+            [fluree.db.storage.ipfs :as ipfs-store]
+            [fluree.db.serde.json :refer [json-serde]]
+            [fluree.db.nameservice.storage-backed :as storage-nameservice]
+            [fluree.db.nameservice.ipns :as ipns-nameservice]
+            [fluree.db.flake.index.storage :as index.storage]
+            #?(:clj  [fluree.db.storage.s3 :as s3-store]
+               :cljs [fluree.db.storage.localstorage :as localstorage-store])
+            [integrant.core :as ig]))
+
+(derive :fluree.storage/file :fluree/content-storage)
+(derive :fluree.storage/file :fluree/byte-storage)
+(derive :fluree.storage/file :fluree/json-archive)
```

Comment:
What's the semantic distinction between these types of storage? `content-storage` and `byte-storage` I get, but what's a `json-archive`?

Reply:
json-archive corresponds to the `fluree.db.storage/JsonArchive` protocol. It's a read-only protocol with a single method for retrieving json data. It's the other side of the content-addressed store protocol, which only has a method for writing data. most of the storage implementations implement both protocols, but the remote storage implementation only implements JsonArchive because you can't modify data in a remote store; you can only read it.